docs/library/usocket.rst

```diff
@@ -315,7 +315,8 @@ Methods
 
 .. method:: socket.readline()
 
-   Read a line, ending in a newline character.
+   Read a line, ending in a newline character. It may return sooner if timeout
+   is reached. (This method has no configurable timeout.)
```

Comment:
It looks like you put this in the wrong file, it should go in docs/library/machine.UART.rst.  Also note that the timeout is configurable via init/the UART constructor.